### ISSUE TITLE
OpenMP no benefit of 1 vs 4 cores for IP twin model 

### ISSUE THREAD

**Issue by tbates** [OpenMx/OpenMx#288]
Running an IP twin model, there's no speed up for 4 cores vs. 1 cores

1. Setup data
```R
data(GFF)
mzData <- subset(GFF, zyg_2grp == "MZ")
dzData <- subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff","fc","qol","hap","sat","AD") # These will be expanded into "gff_T1" "gff_T2" etc.
```

2. Demonstrate that cores have no effect 
```R
umx_set_cores(1); m1 =    umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData); umx_time(m1)
# IPa1c1e1: 07.24 sec
umx_set_cores(4); m1 =    umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData); umx_time(m1)
# IPa1c1e1: 06.55 sec (expected under 2 seconds)

```

All this on Mac 4/8 core laptop

> OpenMx version: 2.17.4 [GIT v2.17.4]
R version: R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu

**Comment by RMKirkpatrick** [comment]
I reproduce under Debian GNU/Linux running on an Intel Core 2 Duo:
```
umx_set_cores(1); m1 =    umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData); umx_time(m1)
# IPa1c1e1: 16.81 seconds
umx_set_cores(2); m1 =    umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData); umx_time(m1)
# IPa1c1e1: 17.41 seconds
```

My `mxVersion()`:
```
OpenMx version: 2.17.3.181 [GIT v2.17.3-181-g34548e58-dirty]
R version: R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu
Default optimizer: SLSQP
NPSOL-enabled?: Yes
OpenMP-enabled?: Yes
```

**Comment by tbates** [comment]
Here it is with diagnostics

**TLDR**: looks like this is purely an effect of the default `rowwiseParallel` being TRUE, which yields zero benefit of extra cores.

> 1 cores  rowFALSE: 08.38 sec rowTRUE: 08.67 sec
> 4 cores rowFALSE: 02.69 sec; rowTRUE: 08.28 sec = 3x speedup

1. Setup inc. diagnostics.

```R
mxOption(key="Parallel diagnostics", value = "Yes")
data(GFF)
mzData <- subset(GFF, zyg_2grp == "MZ")
dzData <- subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff","fc","qol","hap","sat","AD") # These will be expanded into "gff_T1" "gff_T2" etc.

2. Run with 4 cores

```R
umx_set_cores(4);
m0 = umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData, autoRun = FALSE);
mRowFALSE = mxRename(m0, "rowFALSE")
mRowTRUE  = mxRename(m0, "rowTRUE")
# fast
mRowFALSE$MZ$fitfunction$rowwiseParallel = FALSE
mRowFALSE$DZ$fitfunction$rowwiseParallel = FALSE
# slow
mRowTRUE$MZ$fitfunction$rowwiseParallel = TRUE
mRowTRUE$DZ$fitfunction$rowwiseParallel = TRUE

mRowTRUE  = mxRun(mRowTRUE)
mRowFALSE  = mxRun(mRowFALSE)

umx_time(c(mRowFALSE, mRowTRUE))
```R

output for 4 core run

```R
mRowTRUE  = mxRun(mRowTRUE)
Running rowTRUE with 42 parameters
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
FitContext::createChildren: MZ.fitfunction is an OpenMP user
FitContext::createChildren: DZ.fitfunction is an OpenMP user
FitContext::createChildren: create 4 FitContext for parallel processing; OpenMP user=1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MxComputeGradientDescent(SLSQP) evaluations 9801 fit 150679 change -0.0001109MZ.fitfunction: rowwiseParallel used 1 threads; recommend rowwiseParallel=FALSE
DZ.fitfunction: rowwiseParallel used 1 threads; recommend rowwiseParallel=FALSE
JacobianGadget: not used (analytic?)
Gradient used 1/1 threads for 42 free parameters
FitContext::createChildren: MZ.fitfunction is an OpenMP user
FitContext::createChildren: DZ.fitfunction is an OpenMP user
FitContext::createChildren: create 4 FitContext for parallel processing; OpenMP user=1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
                                                                             
> mRowFALSE  = mxRun(mRowFALSE)
Running rowFALSE with 42 parameters
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
FitContext::createChildren: create 4 FitContext for parallel processing; OpenMP user=0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
[0] gradient took 10.731068ms with 4 threads and 14.064222ms with 3 threads
[0] gradient: looks like 4 threads offer the best performance
MxComputeGradientDescent(SLSQP) evaluations 7823 fit 150679 change -0.01075JacobianGadget: not used (analytic?)
Gradient used 4/4 threads for 42 free parameters
FitContext::createChildren: create 4 FitContext for parallel processing; OpenMP user=0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
                                                                            
umx_time(c(mRowFALSE, mRowTRUE))
rowFALSE: 02.69 sec , 
rowTRUE: 08.28 sec (Δ: 5.595), 307.938%
```

3. test with 1 core


```R
umx_set_cores(1);
m0 = umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData, autoRun = FALSE);
mRowFALSE = mxRename(m0, "rowFALSE")
mRowTRUE  = mxRename(m0, "rowTRUE")
# fast
mRowFALSE$MZ$fitfunction$rowwiseParallel = FALSE
mRowFALSE$DZ$fitfunction$rowwiseParallel = FALSE
# slow
mRowTRUE$MZ$fitfunction$rowwiseParallel = TRUE
mRowTRUE$DZ$fitfunction$rowwiseParallel = TRUE

mRowTRUE  = mxRun(mRowTRUE)
mRowFALSE  = mxRun(mRowFALSE)

umx_time(c(mRowFALSE, mRowTRUE))
```R


output from 1 core run

```R
> mRowTRUE  = mxRun(mRowTRUE)
Running rowTRUE with 42 parameters
MZ.fitfunction: rowwiseParallel = 1
DZ.fitfunction: rowwiseParallel = 1
FitContext::createChildren: max threads set to 1
MxComputeGradientDescent(SLSQP) evaluations 10232 fit 150690 change -0.0003256MZ.fitfunction: rowwiseParallel used 1 threads; recommend rowwiseParallel=FALSE
DZ.fitfunction: rowwiseParallel used 1 threads; recommend rowwiseParallel=FALSE
JacobianGadget: not used (analytic?)
Gradient used 1/1 threads for 42 free parameters
FitContext::createChildren: max threads set to 1
Warning message:                                                              
In model 'rowTRUE' Optimizer returned a non-zero status code 5. The Hessian at the solution does not appear to be convex. See ?mxCheckIdentification for possible diagnosis (Mx status RED). 
> mRowFALSE  = mxRun(mRowFALSE)
Running rowFALSE with 42 parameters
MZ.fitfunction: rowwiseParallel = 0
DZ.fitfunction: rowwiseParallel = 0
FitContext::createChildren: max threads set to 1
MxComputeGradientDescent(SLSQP) evaluations 9114 fit 150690 change -0.0002931JacobianGadget: not used (analytic?)
Gradient used 1/1 threads for 42 free parameters
FitContext::createChildren: max threads set to 1
Warning message:                                                             
In model 'rowFALSE' Optimizer returned a non-zero status code 5. The Hessian at the solution does not appear to be convex. See ?mxCheckIdentification for possible diagnosis (Mx status RED). 
> 
> umx_time(c(mRowFALSE, mRowTRUE))
rowFALSE: 08.38 sec , 
rowTRUE: 08.67 sec (Δ: 0.283), 103.37%

```
